consensus/src/quorum_store/batch_generator.rs

```diff
@@ -161,23 +163,35 @@ impl BatchGenerator {
         true
     }
 
+    fn get_descending_bucket_key(&self, txn: &SignedTransaction) -> u64 {
+        let bucket_index = match self.buckets.binary_search(&txn.gas_unit_price()) {
+            Ok(bucket_start) => bucket_start as u64,
+            Err(bucket_start) => bucket_start as u64,
+        };
+        u64::MAX - bucket_index
+    }
+
     fn bucket_into_batches(
         &mut self,
         pulled_txns: &mut Vec<SignedTransaction>,
         expiry_time: u64,
     ) -> Vec<Batch> {
-        pulled_txns.sort_by_key(|txn| txn.gas_unit_price());
+        // Sort by gas bucket, in descending order. Due to stable sort on existing mempool account
+        // ordering, this will result in (descending(gas), account) order.
+        pulled_txns.sort_by_cached_key(|txn| self.get_descending_bucket_key(txn));
```

Comment:
why not just sort by u64::max - gas_unit_price?

Reply:
Makes sense, since this is descending order. 👍 
I think I was playing with ascending order at some point, where the bucketing would have to be used.

Reply:
I think it's even simpler to sort_by(|a, b| b.gas_unit_price().cmp(a.gas_unit_price())?